src/nbl/asset/utils/CHLSLCompiler.cpp

```diff
@@ -25,22 +25,186 @@
 #include <boost/wave/cpplexer/cpp_lex_token.hpp>
 #include <boost/wave/cpplexer/cpp_lex_iterator.hpp>
 
-#define TCPP_IMPLEMENTATION
-#include <tcpp/source/tcppLibrary.hpp>
-#undef TCPP_IMPLEMENTATION
-
 using namespace nbl;
 using namespace nbl::asset;
 using Microsoft::WRL::ComPtr;
 
 static constexpr const wchar_t* SHADER_MODEL_PROFILE = L"XX_6_7";
 
+
+
 namespace nbl::asset::hlsl::impl
 {
-    struct DXC {
+    struct DXC 
+    {
         ComPtr<IDxcUtils> m_dxcUtils;
         ComPtr<IDxcCompiler3> m_dxcCompiler;
     };
+
+
+    // for including builtins 
+    struct load_file_or_builtin_to_string
+    {
+
+    template <typename IterContextT>
+    class inner
+    {
+    public:
+        template <typename PositionT>
+        static void init_iterators(IterContextT& iter_ctx, PositionT const& act_pos, boost::wave::language_support language)
+        {
+            using iterator_type = typename IterContextT::iterator_type;
+
+            std::string filepath(iter_ctx.filename.begin(), iter_ctx.filename.end());
+            auto inclFinder = iter_ctx.ctx.get_hooks().m_preprocessOptions.includeFinder;
+            if (inclFinder) 
+            {
+                std::optional<std::string> result;
+                system::path requestingSourceDir(iter_ctx.ctx.get_current_directory().string());
```

Comment:
needs a total override of their CWD handling, note the behaviour of our old code
https://github.com/Devsh-Graphics-Programming/Nabla/pull/608#discussion_r1399297717

Reply:
no that `requestingSourceDir` was empty (root) for builtins and system paths